packages/core-plugin-api/src/analytics/useAnalytics.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useAnalyticsDomain } from './AnalyticsDomain';
+import {
+  analyticsApiRef,
+  AnalyticsTracker,
+} from '../apis/definitions/AnalyticsApi';
+import { useApi } from '../apis';
+
+function useTracker(): AnalyticsTracker {
+  const analyticsApi = useApi(analyticsApiRef);
+  const domain = useAnalyticsDomain();
+  return {
+    captureEvent: (verb, noun, value, context) => {
+      try {
+        analyticsApi.captureEvent({
+          verb,
+          noun,
+          value,
+          context,
+          domain,
+        });
+      } catch {
+        // Just don't throw on an instrumentation problem.
+      }
+    },
+  };
+}
+
+/**
+ * Get a pre-configured analytics tracker.
+ */
+export function useAnalytics(): AnalyticsTracker {
+  // Return a no-op tracker if no implementation for the Analytics API is
+  // available. Having no default Analytics API implementation enables simple
+  // provider installation via plugin instantiation.
```

Comment:
Thinking that we should just have a default no-op `AnalyticsApi` registered in the app instead of this workaround. Also a bit worried about catching the hook error as we've about to allow APIs to be provided after the initial load.

Reply:
>  _We're about to allow APIs to be provided after the initial load._

👀  Interesting!  What's the use-case?

I ended up going down this route because, I'd really like the install for analytics plugins to be as simple as possible (e.g. "just add `export { someAnalyticsModule } from '@backstage/plugin-analytics-module-some'` to `app/src/plugins.ts` and you're off to the races!"

But this style of installation is not possible right now with a default, no-op implementation registered in the app.  ...Will this _post-load_ API providing allow for this use-case?

Reply:
It's for a more SaaS oriented Backstage deployment where you can load in plugins more lazily.

What do you think about this installation method? It's a very similar amount of code but also allows more easier customization through code if needed: https://github.com/backstage/backstage/blob/58633d0762995c160743d16d9ebac4dd6bc04bc2/.changeset/brave-shoes-push.md

Reply:
It's more than a one-liner, but it's fine.